enhancements/cluster-logging/elasticsearch-v2-api-and-infrastructure.md

```diff
@@ -0,0 +1,319 @@
+  rename the PVCs and rebind the backing PV so that it matches the updated naming convention).
+
+### Risks and Mitigations
+
+## Not Breaking Other Operators who Depend on EO
+
+As part of adding another version of the CRD that EO will respond to, we need to ensure that we can continue to support those that remain on V1 so that
+they are able to migrate their usage to V2 in a timely manner that suits them (typically we give 1-2 releases since announcement before deprecating).
+
+## Design Details
+
+### PVC current and proposed
+
+Current storage PVCs + backing:
+```bash
+$ oc get pvc
+NAME                                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+elasticsearch-elasticsearch-cd-w5afmzhp-1   Bound    pvc-b7c88b5e-0513-4912-8d13-0d4b81e20fe4   10Gi       RWO            gp2            13m
+elasticsearch-elasticsearch-cd-w5afmzhp-2   Bound    pvc-f3193fc1-da91-4267-b819-3b7b0af76cd6   10Gi       RWO            gp2            13m
+elasticsearch-elasticsearch-cd-w5afmzhp-3   Bound    pvc-2936caa5-9311-4548-b3ce-7a2ec5c439ea   10Gi       RWO            gp2            13m
+elasticsearch-elasticsearch-m-xpk5ybty      Bound    pvc-c90afaab-b4d4-47ff-8dd6-15f411ebe999   10Gi       RWO            gp2            13m
+```
+
+Proposed storage PVCs + backing:
+```bash
+$ oc get pvc
+NAME                      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+elasticsearch-0           Bound    pvc-b7c88b5e-0513-4912-8d13-0d4b81e20fe4   10Gi       RWO            gp2            13m
+elasticsearch-1           Bound    pvc-f3193fc1-da91-4267-b819-3b7b0af76cd6   10Gi       RWO            gp2            13m
+elasticsearch-2           Bound    pvc-2936caa5-9311-4548-b3ce-7a2ec5c439ea   10Gi       RWO            gp2            13m
+elasticsearch-master      Bound    pvc-c90afaab-b4d4-47ff-8dd6-15f411ebe999   10Gi       RWO            gp2            13m
+```
+
+### T-Shirt sizing
+
+To take inspiration from EC2 node sizing and the Loki Operator the Elasticsearch Operator would provide a more simple interface to customers
+in the form of t-shirt sizes that would encapsulate replication policies, range of node count, resources provided per node.
+
+This would need to be further fleshed out in a spike to determine what a Medium (the 90% use case) would be sized at, however some example sizes
+could be:
+
+* 0x.Small - A small (e.g. Dev test cluster) with zero redundancy. This would likely only be able to support a small amount of retention.
+  * This could be comprised of just 1 master node and 1-3 data+ingest nodes with 1:1 separate coordinating nodes (matching data nodes)
+* 1x.Medium - A cluster that should be able to cover 90% of our customer use cases and provide single redundancy
+  * 3 masters, 3-5 data+ingest nodes and 1:1 coordinating nodes
+* 1x.Large - For very large customer clusters with single redundancy
```

Comment:
How does the naming convention apply to `MultipleRedudancy` and `FullRedudancy` as supported today?

Reply:
these are just example names, not ones that are set in stone. its up to the implementation to decide that